gadopt/stokes_integrators.py

```diff
@@ -97,33 +97,32 @@ def create_stokes_nullspace(Z, closed=True, rotational=False, translations=None)
 class StokesSolver:
     name = 'Stokes'
 
-    def __init__(self, z, T, approximation, bcs=None, mu=1,
-                 quad_degree=6, cartesian=True, solver_parameters=None,
-                 closed=True, rotational=False, J=None, constant_jacobian=False,
-                 **kwargs):
+    def __init__(self, z, T, approximation, bcs=None, mu=1, quad_degree=6,
+                 cartesian=True, solver_parameters=None, closed=True,
+                 rotational=False, J=None, constant_jacobian=False, **kwargs):
+
         self.Z = z.function_space()
         self.mesh = self.Z.mesh()
         self.test = fd.TestFunctions(self.Z)
-        self.equations = StokesEquations(self.Z, self.Z, quad_degree=quad_degree,
-                                         compressible=approximation.compressible)
         self.solution = z
+        self.solution_old = fd.Function(self.solution)  # This is used for the implicit free surface coupling
+        self.T = T
         self.approximation = approximation
         self.mu = ensure_constant(mu)
+        self.quad_degree = quad_degree
         self.solver_parameters = solver_parameters
         self.J = J
         self.constant_jacobian = constant_jacobian
         self.linear = not depends_on(self.mu, self.solution)
-
         self.solver_kwargs = kwargs
-        u, p = fd.split(self.solution)
```

Comment:
change this to  `solutions_split = fd.split(self.solution)` instead of dropping it.

Reply:
I've changed this now to only take the first two elements of split.

Reply:
actually changed it back to how you suggested!